common/tools/dev-tool/src/config/rollup.base.config.ts

```diff
@@ -0,0 +1,118 @@
+  return warning.code === "EVAL" && (
+    warning.id?.includes("node_modules/nise") ||
+    warning.id?.includes("node_modules/sinon")) === true;
+}
+
+function ignoreChaiCircularDependencyWarnings(warning: RollupWarning): boolean {
+  return warning.code === "CIRCULAR_DEPENDENCY" &&
+      warning.importer?.includes("node_modules/chai") === true;
+}
+
+const warningInhibitors: Array<(warning: RollupWarning) => boolean> = [
+  ignoreChaiCircularDependencyWarnings,
+  ignoreNiseSinonEvalWarnings
+];
+
+/**
+ * Construct a warning handler for the shared rollup configuration
+ * that ignores certain warnings that are not relevant to testing.
+ */
+function makeOnWarnForTesting(): (warning: RollupWarning, warn: WarningHandler) => void {
+  return (warning, warn) => {
+    // If every inhibitor returns false (i.e. no inhibitors), then show the warning
+    if (warningInhibitors.every((inhib) => !inhib(warning))) {
+      warn(warning);
+    }
+  }
+}
+
+// #endregion
+
+function makeBrowserTestConfig() {
+  const config: RollupOptions = {
+    input: ["dist-esm/test/{,!(node)/**/}*.spec.js"],
+    output: {
+      file: `dist-test/index.browser.js`,
+      format: "umd",
+      sourcemap: true,
+      globals: { "fs-extra": "undefined" }
+    },
+    preserveSymlinks: false,
+    // fs-extra must be marked as external in order to avoid an initialization error
+    external: ["fs-extra"],
+    plugins: [
+      multiEntry({ exports: false }),
+      nodeResolve({
+        mainFields: ["module", "browser"],
+        preferBuiltins: true
+      }),
+      cjs({
+        namedExports: {
+          "chai": ["assert", "use"],
+          "@opentelemetry/api": ["CanonicalCode", "SpanKind", "TraceFlags"]
+        }
+      }),
+      json(),
+      sourcemaps(),
+      nodeGlobals(),
+      nodeBuiltinsPlugin()
+      //viz({ filename: "dist-test/browser-stats.html", sourcemap: true })
+    ],
+    onwarn: makeOnWarnForTesting(),
+    // Disable tree-shaking of test code.  In rollup-plugin-node-resolve@5.0.0, rollup started respecting
+    // the "sideEffects" field in package.json.  Since our package.json sets "sideEffects=false", this also
+    // applies to test code, which causes all tests to be removed by tree-shaking.
+    treeshake: false
+  };
+
+  // (config.external as string[]).push(...Object.keys(pkg.devDependencies));
+
+  return config;
+}
+
+export function makeConfig(pkg: PackageJson)  {
+  const baseConfig = {
+    input: "dist-esm/src/index.js",
+    external: [...nodeBuiltins, ...Object.keys(pkg.dependencies), ...Object.keys(pkg.devDependencies)],
+    output: { file: "dist/index.js", format: "cjs", sourcemap: true },
+    preserveSymlinks: false,
+    plugins: [sourcemaps(), nodeResolve(), cjs()]
+  }
+
+  if (IS_PRODUCTION) {
+    baseConfig.plugins.push(terser());
+  }
+
+  const config: RollupOptions[] = [baseConfig as RollupOptions];
+
+  if (!IS_PRODUCTION) {
+    config.push(makeBrowserTestConfig());
```

Comment:
I'm not sure why this is gated on the production env var

Reply:
to me the logic was about whether or not we were building a production-ready distribution bundle, and in that case it seemed like we wouldn't need to generate the browser testing bundle, but honestly I don't know if/where we use NODE_ENV=production in the pipelines. I was expecting that I would see a CI failure if the browser test bundle was needed in any production contexts, but since I didn't...

I'll need to investigate this a little further.

Reply:
Thought about it and I have come around to the same point of view on both this and the other IS_PRODUCTION comment. We don't need to do any minifying and we can just build the browser bundle by default. In any case, the NODE_ENV is the wrong switch to use here. I will instead add an options parameter with an opt-out flag for those packages that want to disable the browser test build.